libs/auth/src/openssl.ts

```diff
@@ -186,68 +211,254 @@ export async function verifySignature({
   ]);
 }
 
+//
+// Cert creation functions
+//
```

Comment:
There's quite a lot going on in this file, so here's a quick overview for any reviewers not as familiar with cert creation!

-----

Cert creation involves two steps under the hood: 1) creation of a cert signing request, or CSR, and 2) creation of a cert from that CSR.

A common flow is as follows:

- Entity 1 creates a CSR with its private key.
- Entity 1 shares the CSR with entity 2. The CSR is safe to share in public.
- Entity 2 signs the CSR with its private key to create a cert.
- Entity 2 shares the cert with entity 1. The cert is safe to share in public.

The end result is a cert containing entity 1's public key signed by entity 2's private key.

-----

When programming smart cards, we're performing all of the above on a single machine. So we can combine all of the above into a single function. But I've updated the code to better distinguish between the two steps. The separation will make it easier for me to use this code for other purposes, like machine cert creation, which does require two machines.

An important caveat for smart card programming is that we can't access/use Java Card private keys for CSR creation. Thankfully, it's still possible to create certs using just entity 1's public key. This involves using a throwaway private key and the `-force_pubkey` param, which you'll see in the code.

Lastly, using the TPM key in prod requires sudo access. There's a slightly roundabout control flow to accommodate this, described in the comments for `createCert`.